config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get 'posts/index'
+  get 'posts/show'
+  root 'users#index'
+  get 'users', to: 'users#index'
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
### OPTIONAL 

- [ ] It would be better to create your routes like this below ⬇️ 
```ruby
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```
- The above looks more structured and you loop through users to see posts they created